clients/go/cipher.go

```diff
@@ -205,54 +186,58 @@ func (c X25519DeoxysIICipher) EncryptEncode(plaintext []byte) []byte {
 }
 
 func (c X25519DeoxysIICipher) DecryptCallResult(response []byte) ([]byte, error) {
-	var callResult CallResult
-	cbor.MustUnmarshal(response, &callResult)
+	var callResult types.CallResult
+	if err := cbor.Unmarshal(response, &callResult); err != nil {
+		return nil, err
+	}
 
 	// TODO: actually decode and return failure
-	if callResult.Fail != nil {
+	if callResult.Failed != nil {
 		return nil, ErrCallFailed
 	}
 
-	var aeadEnvelope AeadEnvelope
-	if callResult.OK != nil {
-		if err := cbor.Unmarshal(callResult.OK, &aeadEnvelope); err != nil {
-			return callResult.OK, nil
+	var aeadEnvelope types.ResultEnvelopeX25519DeoxysII
```

Comment:
Same here. I would probably expect that cipher defines an envelope, not an external library.